### ISSUE TITLE
RHF 7 - useFieldArray Returns Wrong Type For "Fields" For Optional Property

### ISSUE THREAD

**Issue by dudiharush** [react-hook-form/react-hook-form#4693]
**Describe the bug**
When I choose a value for name prop at useFieldArray function that has an interface with optional type, the returned "fields" param gets a `never[]` type instead of `undefined | Item`

**To Reproduce**

1. Go to [codesandbox example](https://codesandbox.io/s/admiring-feather-olshf?file=/src/App.tsx)
2. See error
`Property 'name' does not exist on type 'never'.ts(2339)`

V7
[codesandbox example](https://codesandbox.io/s/admiring-feather-olshf?file=/src/App.tsx)

**Expected behavior**
I would expect fields to have `undefined | Item[]` type

**Comment by bluebill1049** [comment]
you didn't pass the generate to the useForm: https://codesandbox.io/s/restless-water-dk2k2

**Comment by rvashchenko** [comment]
@bluebill1049 in your example, if you make the items field optional in the 'SomeObject' type (items ?: Item [];), problems arise with the types of array elements. In particular, with the returned: fields, append, prepend, insert. fields is of the wrong type 'never []' instead 'Item[]',  append, prepend, insert arguments require data in the wrong format 'never []', instead of 'Item' or 'Item[]'

**Comment by magdalko** [comment]
We have exactly the same problem! Using equivalent of SomeObject as optional which results in fields being of type never[]. @bluebill1049 @rvashchenko

**Comment by bluebill1049** [comment]
can I have a simple codesandbox to reproduce the issue?

**Comment by theJasonHall** [comment]
@bluebill1049  I ran into a similar issue to this, so I went ahead and created two codesandboxes.  One demonstrates this issue and the other demonstrates my issue.  Both are optional properties showing as `never`.

This issue (useFieldArray):  https://codesandbox.io/s/fast-currying-5hp5x?file=/src/App.tsx
My issue (setValue):  https://codesandbox.io/s/bitter-leftpad-74ofs?file=/src/App.tsx

I'm thinking this might be a limitation of TypeScript.  In the second codesandbox, I'm not able to do something like the following:
```
type test = SomeObject['item']['name'];
```
